frontend/components/Editor/LaunchBackendButton.js

```diff
@@ -0,0 +1,45 @@
+import { html, useMemo, useEffect } from "../../imports/Preact.js"
+import * as preact from "../../imports/Preact.js"
+import { RunLocalButton, BinderButton } from "../EditOrRunButton.js"
+import { start_local } from "../../common/RunLocal.js"
+import { BackendLaunchPhase, start_binder } from "../../common/Binder.js"
+import immer, { applyPatches, produceWithPatches } from "../../imports/immer.js"
+
+export const EditorLaunchBackendButton = ({ editor, launch_params, status }) => {
+    try {
+        const EnvRun = useMemo(
+            // @ts-ignore
+            () => window?.pluto_injected_environment?.environment?.({ client: editor.client, editor, imports: { immer, preact } })?.custom_run_or_edit,
+            [editor.client, editor]
+        )
+        // @ts-ignore
+        if (window?.pluto_injected_environment?.provides_backend) {
+            // @ts-ignore
+            return html`<${EnvRun} editor=${editor} backend_phases=${BackendLaunchPhase} launch_params=${launch_params} />`
+            // Don't allow a misconfigured environment to stop offering other backends
+        }
+    } catch (e) {}
```

Comment:
Are you sure you don't want to log this?

Reply:
Currently I don't; there is no easy way to get these logs from clients and these show a setup issue - not a runtime one; should be fixed in "staging". Also there will be a failed network request along this.